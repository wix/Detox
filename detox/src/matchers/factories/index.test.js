describe('Matchers factories', () => {
  let factories;
  beforeEach(() => {
    factories = require('./index');
  });

  describe('External-module', () => {
    describe('given module with missing matchers class', () => {
      it('should throw an error', () => {
        const module = {
          ExpectClass: undefined,
        };
        expect(() => factories.External.validateModule(module, 'fake/path')).toThrowErrorMatchingSnapshot();
      });
    });

    describe('given proper module', () => {
      it('should not throw an error', () => {
        const module = {
          ExpectClass: class {},
        };
        factories.External.validateModule(module, 'fake/path');
      });
    });
  });
});
