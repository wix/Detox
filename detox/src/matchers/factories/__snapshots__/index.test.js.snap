// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matchers factories External-module given module with missing matchers class should throw an error 1`] = `"The custom driver at 'fake/path' does not export the ExpectClass property"`;
